opensearchpy/connection/http_requests.py

```diff
@@ -244,7 +252,15 @@ def perform_request(  # type: ignore
             duration,
         )
 
-        return response.status_code, response.headers, raw_data
+        if self.metrics is None:
+            return response.status_code, response.headers, raw_data
+        else:
+            return (
+                response.status_code,
+                response.headers,
+                raw_data,
+                self.metrics.service_time,
+            )
```

Comment:
I don't think you should be returning any of the metrics, the caller can look into `.metrics.service_time` if they need to.